hopp/mhk_wave_source.py

```diff
@@ -0,0 +1,342 @@
+        system_model.MHKWave.significant_wave_height = self.site.wave_resource.data['significant_wave_height'] #significant_wave_height #array, length 2920
+        system_model.MHKWave.energy_period = self.site.wave_resource.data['energy_period'] #energy_period #array, length 2920
+        system_model.MHKWave.year = self.site.wave_resource.data['year'] # year
+        system_model.MHKWave.month = self.site.wave_resource.data['month'] # month
+        system_model.MHKWave.day = self.site.wave_resource.data['day'] #day
+        system_model.MHKWave.hour = self.site.wave_resource.data['hour'] #hour
+        system_model.MHKWave.minute = self.site.wave_resource.data['minute'] #minute
+        # if 'layout_mode' not in mhk_config.keys():
+        #     layout_mode = 'grid'
+        # else:
+        #     layout_mode = mhk_config['layout_mode']
+
+        if 'device_rating_kw' not in mhk_config.keys():
+            raise ValueError("'device_rating_kw' for MHKWavePlant")
+
+        if 'num_devices' not in mhk_config.keys():
+            raise ValueError("'num_devices' required for MHKWavePlant")
+
+        if 'wave_power_matrix' not in mhk_config.keys():
+            raise ValueError("'wave_power_matrix' required for MHKWavePlant")
+
+        # System parameter inputs
+        self.device_rated_power = mhk_config['device_rating_kw']
+        self.number_devices = mhk_config['num_devices']
+        self.power_matrix = mhk_config['wave_power_matrix']
+
+        
+        self._system_model.value("device_rated_power", self.device_rated_power)
+        self._system_model.value("number_devices",  self.number_devices)
+        self._system_model.value("wave_power_matrix", self.power_matrix)
+
+
+        # Losses
+        if 'loss_array_spacing' not in mhk_config.keys():
+            self._system_model.MHKWave.loss_array_spacing = 0
+        else:
+            self._system_model.MHKWave.loss_array_spacing = mhk_config['loss_array_spacing']
+        if 'loss_downtime' not in mhk_config.keys():
+            self._system_model.MHKWave.loss_downtime = 0
+        else:
+            self._system_model.MHKWave.loss_downtime = mhk_config['loss_downtime'] 
+        if 'loss_resource_overprediction' not in mhk_config.keys():
+            self._system_model.MHKWave.loss_resource_overprediction = 0
+        else:
+            self._system_model.MHKWave.loss_resource_overprediction = mhk_config['loss_resource_overprediction'] 
+        if 'loss_transmission' not in mhk_config.keys():
+            self._system_model.MHKWave.loss_transmission = 0
+        else:
+            self._system_model.MHKWave.loss_transmission = mhk_config['loss_transmission']
+        if 'loss_additional' not in mhk_config.keys():
+            self._system_model.MHKWave.loss_additional = 0
+        else:
+            self._system_model.MHKWave.loss_additional = mhk_config['loss_additional']
+
+
+        # Cost parameter inputs
+        self.ref_model_num = cost_model_inputs['reference_model_num']
```

Comment:
Make sure to add docstrings in the class for the cost model inputs. Do you think it'll make sense to have default values, like the ones below, or should the user be expected to set them?

```
	'reference_model_num':3,
	'water_depth': 100,
	'distance_to_shore': 80,
    'number_rows': 10,
	'devices_per_row':10,
	'device_spacing':600,
	'row_spacing': 600,
	'cable_system_overbuild': 20
```

Reply:
I've created a seperate class for the mhk cost calculator. https://github.com/NREL/HOPP/pull/133/commits/9a978b474b029d3ee2002897520a4143320440d6

I think it's better to have the user set the cost inputs for the wave model personally because I think without additional documentation on how to use hopp it becomes easy to miss stuff that's being set to a default value. I was thinking an example input file for wave would be a good supplement.

Looking through the calculate_installed_cost it's taking into account BOS costs and it looks like those are from specific scenarios "greenfield" and "solar addition". I have a different project where I'm going to reconfigure the BOS models in hopp and I think it would make more sense to add wave to the calculate_installed_costs then since it doesn't easily plug-in to the current framework. Thoughts?